Language/Haskell/GhcMod/HomeModuleGraph.hs

```diff
@@ -206,17 +208,33 @@ updateHomeModuleGraph' env smp0 = do
            mapM_ go (Set.toList smp)
 
    step :: ModulePath -> m (Maybe (Set ModulePath))
-   step mp = runMaybeT $ do
-       (dflags, ppsrc_fn) <- MaybeT preprocess'
-       src <- liftIO $ readFile ppsrc_fn
-       imports mp src dflags
-    where
-      preprocess' :: m (Maybe (DynFlags, FilePath))
-      preprocess' = do
+   step mp = do
         let fn = mpPath mp
-        ep <- liftIO $ withLogger' env $ \setDf -> let
+        mapping <- lookupMMappedFile fn
+        runMaybeT $ do
+          let preprocess'' fn' = do
+                (dflags, ppsrc_fn) <- MaybeT (preprocess' fn')
+                src <- liftIO $ readFile ppsrc_fn
+                return (dflags, src)
+          (dflags, src) <- case mapping of
+            Just (RedirectedMapping to) -> preprocess'' to
+            Just (MemoryMapping (Just src)) -> do
+              (fp, hndl) <- liftIO $ getTemporaryDirectory >>= \tmpDir' ->
```

Comment:
Actually, getTemporaryDirectory just returns path to `/tmp` or `C:\Temp` or something. But I get your point, better use cradleTempDir anyway.